### PR TITLE
Update MQTT config for Keep Alive demo that uses dummy timer

### PR DESCRIPTION
**Pull request by aggarw13**
The MQTT library has been updated with the `MQTT_RECV_POLLING_TIMEOUT_MS` and `MQTT_SEND_RETRY_TIMEOUT_MS` configurations which should be set to zero when using a dummy timer function to avoid possibility of infinite loop when retrying failed transport send/receive calls. As the MQTT Keep Alive demo uses a dummy timer function, this PR updates the demo configuration to set these timeouts to zero.

### CODE REVIEW

**Review by yourslab — DISMISSED** [review]

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h [hunk 1]
```diff
@@ -75,6 +74,41 @@ extern void vLoggingPrintf( const char * pcFormatString,
...
+ *
+ * When an incoming MQTT packet is detected, the transport receive function
+ * may be called multiple times until all of the expected number of bytes of the
+ * packet are received. This timeout represents the maximum polling duration that
+ * is allowed without any data reception from the network for the incoming packet.
+ *
+ * @note For this demo, the timeout value is configured to zero as the demo uses a
+ * dummy timer function (of #MQTTGetCurrentTimeFunc_t) that always returns zero.
+ * It is REQUIRED to set the the timeout to zero when using a dummy timer function
+ * that always zero.
```
Comment (dan4thewin) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:92]: Should that say "always returns zero"?

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h [hunk 2]
```diff
@@ -75,6 +74,41 @@ extern void vLoggingPrintf( const char * pcFormatString,
...
+ */
+#define MQTT_RECV_POLLING_TIMEOUT_MS    0U
+
+/**
+ * @brief The maximum duration between non-empty network transmissions while
+ * sending an MQTT packet via the #MQTT_ProcessLoop or #MQTT_ReceiveLoop
+ * API functions.
+ *
+ * When sending an MQTT packet, the transport send function may be called multiple
+ * times until all of the required number of bytes are sent.
```
Comment (dan4thewin) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:102]: remove "all of"?

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h [hunk 3]
```diff
@@ -75,6 +74,41 @@ extern void vLoggingPrintf( const char * pcFormatString,
...
+ * This timeout represents the maximum duration that is allowed for no data
```
Comment (dan4thewin) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:103]: maximum time to wait for what?

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h (hunk 1 above)
Comment (aggarw13) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:92]:
```suggestion
 * that always returns zero.
```

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h (hunk 2 above)
Comment (aggarw13) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:102]:
```suggestion
 * times until the required number of bytes are sent.
```

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h (hunk 3 above)
Comment (aggarw13) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:103]:
```suggestion
 * This timeout represents the maximum time wait for any data
```

Comment (aggarw13) [reply FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h]: Maximum time to wait for a single byte to be transmitted over the network over multiple calls to the transport send function.

Comment (dan4thewin) [reply FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h]: Can you make the code comment read more like this explanation?

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h [hunk 4]
```diff
@@ -75,6 +74,41 @@ extern void vLoggingPrintf( const char * pcFormatString,
...
+ *
+ * When sending an MQTT packet, the transport send function may be called multiple
+ * times until the required number of bytes are sent.
+ * This timeout represents the maximum time wait for any data
+ * transmission over the network through the transport send function.
+ *
+ * @note For this demo, the timeout value is configured to zero as the demo uses a
+ * dummy timer function (of #MQTTGetCurrentTimeFunc_t) that always returns zero.
+ * It is REQUIRED to set the the timeout to zero when using a dummy timer function
+ * that always zero.
```
Comment (dan4thewin) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:109]: always returns?

Comment (aggarw13) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/core_mqtt_config.h:109]:
```suggestion
 * that always returns zero.
```

**Review by dan4thewin — APPROVED** [review]

**Review by yourslab — APPROVED** [review]

**Pull request merged**